gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dcombobox_ttt2.lua

```diff
@@ -97,33 +126,94 @@ function PANEL:PerformLayout()
 end
 
 ---
--- @param string value
--- @param number index
+-- @param string title
+-- @param string/number value should be indexable
+-- @param bool select
+-- @param string icon
+-- @param any data additional data that you might want to use
+-- @return number index
 -- @realm client
-function PANEL:ChooseOption(value, index)
-	if self.menu then
-		self.menu:Remove()
-		self.menu = nil
+function PANEL:AddChoice(title, value, select, icon, data)
+	if istable(value) then
+		ErrorNoHalt("[TTT2] dcombobox_ttt2 AddChoice format changed to PANEL:AddChoice(title, value, select, icon, data)\n For any table data use the last parameter.\n")
+
+		return
 	end
 
-	self:SetText(value)
+	local index = #self.choices + 1
+
+	self.choices[index] = {
+		title = title,
+		value = value,
+		icon = icon,
+		data = data
+	}
+
+	-- Index tables for fast access
+	self.titleIndices[title] = index
+	self.valueIndices[value] = index
+
+	if select then
+		self:ChooseOption(nil, index, true)
+	end
+
+	return index
+end
+
+---
+-- Choose either value or index, title is set 
+-- @param[opt] string/number value should be indexable
+-- @param[opt] number index the option id
+-- @param[default=false] bool ignoreConVar To avoid endless loops, separated setting of convars and UI values
+-- @realm client
+function PANEL:ChooseOption(value, index, ignoreConVar)
```

Comment:
I don't really like the design on those two parameters here. 
Either use another function specifically for choosing by value which looks up the index and then calls this function with the index or solve it in another way that does not include using two parameters where only one is required, this feels weird.

Just saw that this is the base function for others that already exist.
Please still rewrite them to use eg. the index function as a base and let the others only handle converting the value to the right index.

Reply:
I mainly tried to stick to the original, can remove all old behaviour nonetheless:
https://wiki.facepunch.com/gmod/DComboBox:ChooseOption